LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/MessageDispatcher.cs

```diff
@@ -74,34 +72,15 @@ public void DispatchRequest(LoRaRequest request)
 
             if (request.Payload is null)
             {
-                // Following code is only needed for PktFwd compatibility.
-                // Any LoRaRequest generated from LNS protocol 'updf' or 'jreq' messages already has the payload set.
-                if (!LoRaPayload.TryCreateLoRaPayload(request.Rxpk, out var loRaPayload))
-                {
-                    this.logger.LogError("There was a problem in decoding the Rxpk");
-                    request.NotifyFailed(LoRaDeviceRequestFailedReason.InvalidRxpk);
-                    return;
-                }
-                request.SetPayload(loRaPayload);
+                throw new LoRaProcessingException(nameof(request.Payload), LoRaProcessingErrorCode.PayloadNotSet);
             }
 
-            if (this.loraRegion == null)
+            // I changed the logic here to make the request immediately fails by throwing instead of returning null
```

Comment:
let me know if that is ok

Reply:
@bastbu

Reply:
I don't think I have the necessary context to give a good answer here. Without knowing the history of that code, I wonder if we need the null-check/throw at all? If we don't have the region set in a path, surely we will fail somewhere later in the processing chain, and our tests would catch that (since it's only invoked in the join request and uplink data frame cases, which are covered)? In any case, I would probably remove the comment on L78.